actors/common/src/main/java/org/sunbird/learner/actors/bulkupload/TextbookTocActor.java

```diff
@@ -640,15 +644,18 @@ private void throwInvalidDialCodeError(Set<String> invalidDialCodes) {
       StringBuilder exceptionMsgs = new StringBuilder();
       for (int i = 0; i < csvRecords.size(); i++) {
         CSVRecord record = csvRecords.get(i);
+        Map<String,String> mappingValues=record.toMap();
+        Map<String,String> trimMapping=new HashMap<>();
+        mappingValues.keySet().forEach(key -> trimMapping.put(key.trim(), mappingValues.get(key).trim()));
```

Comment:
try using entrySet, also mappingValues variable is not needed

Map<String,String> trimMapping = record.toMap().entrySet().stream().collect(Collectors.toMap(e->e.getKey().trim(), e->e.getValue().trim()));

do a null check before value trimming

change the 'trimMapping' to 'trimmedRecordMap'